Tools/cases_generator/analyzer.py

```diff
@@ -353,18 +374,43 @@ def analyze_caches(inputs: list[parser.InputEffect]) -> list[CacheEntry]:
     return [CacheEntry(i.name, int(i.size)) for i in caches]
 
 
+def find_assignment_target(node: parser.InstDef, idx: int) -> list[lexer.Token]:
+    """Find the tokens that make up the left-hand side of an assignment"""
+    offset = 0
+    for tkn in reversed(node.block.tokens[: idx]):
+        if tkn.kind == "SEMI" or tkn.kind == "LBRACE" or tkn.kind == "RBRACE":
+            return node.block.tokens[idx - offset : idx]
+        offset += 1
+    return []
+
+
+def find_stores_outputs(node: parser.InstDef) -> list[lexer.Token]:
+    res: list[lexer.Token] = []
+    outnames = [ out.name for out in node.outputs ]
+    innames = [ out.name for out in node.inputs ]
```

Comment:
```suggestion
    outnames = { out.name for out in node.outputs }
    innames = { out.name for out in node.inputs }
```

Those are only used via `__contains__` so we can just use sets.